src/DocChecks.jl

```diff
@@ -366,6 +378,43 @@ function print_indented(buffer::IO, str::AbstractString; indent = 4)
     end
 end
 
+function fix_doctest(result::Result, str, doc::Documents.Document)
+    code = result.block.code
+    filename = Base.find_source_file(result.file)
+    # read the file containing the code block
+    content = open(f -> read(f, String), filename)
+    # output stream
+    io = Compat.IOBuffer(sizehint = sizeof(content))
+    # first look for the entire code block
+    codeidx = Compat.findnext(code, content, 1)
+    if codeidx === nothing
+        Utilities.warn("Could not find code block in source file")
+        return
```

Comment:
This check (and the one below)  _should_ still never happen, right? Perhaps we should keep `throw`ing/`error`ing/`assert`ing here so that in case they are triggered, they don't go unnoticed.

Reply:
My thought here was that since we have replaces `\r\n` with `\n` in the code block https://github.com/JuliaDocs/Documenter.jl/blob/b64a4d7f2200b1a3ca4b051c54eade54bce77bce/src/DocChecks.jl#L133-L134  we would obtain `nothing` here since we have not performed the replacement in the file where we search.